install/helm/agones/templates/crds/_gameserverspecschema.yaml

```diff
@@ -12,22 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{{/* Validation for a gameserver spec */}}
-{{- define "gameserver.validation" }}
+{{/* schema for a gameserver spec */}}
```

Comment:
Question about this file: it looks like it was moved (not copied). Is that going to break compatibility at all? It seems like we should add support for v1 CRDs first, wait a release or two, and then drop support for v1beta1 CRDs (I'm assuming that is possible). 

If not, then this change may make it impossible to do in-place upgrades to the next agones release.

Reply:
> Question about this file: it looks like it was moved (not copied). Is that going to break compatibility at all?

Should make absolutely no difference at all. I just renamed it to be semantically inline with what the details are called in CRD v1.

> It seems like we should add support for v1 CRDs first, wait a release or two, and then drop support for v1beta1 CRDs (I'm assuming that is possible).

That outside of our control. Also the beta definitions of CRDs in Agones are already being converted into v1 CRDs by Kubernetes (in fact, that's how I did some of the initial conversions - looked at the details of how Kubernetes was doing it).

> If not, then this change may make it impossible to do in-place upgrades to the next agones release.

Out e2e cluster seems to be happily doing upgrades from v1 ↔ v1beta1 ↔ v1 again without any concerns, so it looks like in-place upgrades are not an issue.

Reply:
In fact - I just realised something fun to note.

Kubernetes does the conversion in both directions. I.e. the code in Agones still references the CRDs via the beta interface, even though we are installing them as v1.

It's quite a clever little system they have going on in there :+1:

Reply:
Yes, that's part of the reason it took CRDs so long to reach v1.... lots of details to get right. :)